src/ambianic/pipeline/ai/fall_detect.py

```diff
@@ -301,14 +314,16 @@ def find_spinalLine():
 
         return pose_score, pose_dix
 
+
     def fall_detect(self, image=None):
         assert image
         log.debug("Calling TF engine for inference")
         start_time = time.monotonic()
 
         now = time.monotonic()
-        lapse = now - self._prev_time
-        if self._prev_vals and lapse < self.min_time_between_frames:
+        lapse = now - self._prev_time['t1']
```

Comment:
@ivelin Here I have already checked the condition whether we have the frame data at time t1 or not.

[        if self._prev_pose_dix['t1'] and lapse < self.min_time_between_frames:](https://github.com/ambianic/ambianic-edge/blob/9dd818bc01941687630b51dd56e25d380ae7329d/src/ambianic/pipeline/ai/fall_detect.py#L326)

Can you please look at it again?

Reply:
@bhavikapanara Your check will still bomb if there is no element at index ['t1']. 
Read this: https://realpython.com/python-keyerror/

Reply:
@ivelin still I didn't get your point. Please help me by providing an example.

[        if self._prev_data[-1][self.POSE_VAL] and lapse < self.min_time_between_frames:](https://github.com/ambianic/ambianic-edge/blob/5d1abe64d371c71fe9ba66ff9ab615ac7cc9b925/src/ambianic/pipeline/ai/fall_detect.py#L337)

This will check, does frame available at t-1 timestamp or not?